### PR TITLE
v0.4.0

### PR DESCRIPTION
**Pull request by DJ2LS**
### FreedDATA v0.4.0
Some minor protocol changes...

- [x] ~~#171~~ lzib has still higher compression ratio 
- [x] ~~crc check: move from crcengine to crccheck package~~
- [x] introduce callsign encoding created by @DB1UJ
- [x] introduce gridsquare encoding created by @DB1UJ 
- [x] callsign crc: increase 16bit callsign crc to 24bit callsign crc
- [x] Create GUI option for enabling/disabling the QRV response feature
- [x] #170 
- [x] #173 
- [ ] ~~Update CTESTS to sounddevice~~ we need to think about more reliable tests instead of using virtual audio devices
- [x] #179
- [x] Add callsign to frame type 249, sent in `stop_transmission`. Serves as an ID on completion of communications.

### CODE REVIEW

File: gui/sock.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ client.on('error', function(data) {
     ipcRenderer.send('request-update-tnc-state', Data);
     ipcRenderer.send('request-update-tnc-connection', {tnc_connection : client.readyState});
     client.destroy();
-    setTimeout(connectTNC, 500)
+    setTimeout(connectTNC, 1500)
```
[the remark this answers is not in the document]
Comment (kronenpj) [reply gui/sock.js:83]: Would using try/catch with the original timeout be more friendly? I normally see this when the user hasn't connected to the modem daemon yet, so the exception isn't helpful to have scrolling in the window anyway. I have an idea. Let me see if I can get it working.

[the remark this answers is not in the document]
Comment (kronenpj) [reply gui/sock.js]: Committed a change that restores the timeout but doesn't emit more than one message to the console while retrying. Also added "helpful" message on how to fix the error.

**Review by kronenpj — APPROVED** [review]

File: tnc/data_handler.py [hunk 2]
```diff
@@ -1059,15 +1057,18 @@ def open_session(self, callsign):
...
-                    return False
+                        return True
+                # if static.ARQ_SESSION:
+                #     break
+
+            # Session connect timeout, send close_session frame to
+            # attempt to cleanup the far-side, if it received the
+            # open_session frame and can still hear us.
+            if not static.ARQ_SESSION:
+                # if not TESTMODE:
```
[the remark this answers is not in the document]
Comment (DJ2LS) [reply tnc/data_handler.py:1068]: @kronenpj we might check this again later when rethinking the ctests

File: tnc/data_handler.py [hunk 3]
```diff
@@ -1535,8 +1539,9 @@ def stop_transmission(self):
         structlog.get_logger("structlog").warning("[TNC] Stopping transmission!")
         stop_frame      = bytearray(14)
         stop_frame[:1]  = bytes([249])
-        stop_frame[1:3] = static.DXCALLSIGN_CRC
-        stop_frame[3:5] = static.MYCALLSIGN_CRC
+        stop_frame[1:4] = static.DXCALLSIGN_CRC
+        stop_frame[4:7] = static.MYCALLSIGN_CRC
+        stop_frame[7:13]   = helpers.callsign_to_bytes(self.mycallsign)
```
[the remark this answers is not in the document]
Comment (DJ2LS) [reply tnc/data_handler.py:1549]: @kronenpj this is also a point we shouldn't forget. However, we will see this again, when moving to a possible smaller signalling frame when using fsk some when. This shouldnt break the actual protocol

**Comment by DJ2LS** [comment]
@kronenpj I'm going to merge this now as we need to see if the callsign and grid encoding is working well. This is something which needs to be tested well - I will create a new release for this now as its a minor protocol change (0.3.X --> 0.4.0) - and therefore a version increment. We can add later changes without protocol changes at later sub releases ( 0.4.X )

**Pull request merged**